Framework/Algorithms/test/MonteCarloAbsorptionTest.h

```diff
@@ -76,185 +83,495 @@ void addSample(const Mantid::API::MatrixWorkspace_sptr &ws,
-    const double efixed(12.0);
+    const double efixed(wsProps.efixed);
     space->mutableRun().addProperty<double>("Ei", efixed);
   } else if (wsProps.emode == DeltaEMode::Indirect) {
-    const double efixed(1.845);
+    const double efixed(wsProps.efixed);
     pmap.addString(inst.get(), "deltaE-mode", "Indirect");
     pmap.addDouble(inst.get(), "Efixed", efixed);
   }
 
-  addSample(space, wsProps.sampleEnviron, wsProps.beamWidth,
-            wsProps.beamHeight);
+  addSample(space, wsProps.sampleEnviron);
   return space;
 }
 } // namespace
 
 class MonteCarloAbsorptionTest : public CxxTest::TestSuite {
 public:
   //---------------------------------------------------------------------------
-  // Success cases
+  // Integration tests - Success cases
   //---------------------------------------------------------------------------
   void test_Workspace_With_Just_Sample_For_Elastic() {
+    using namespace Mantid::Geometry;
+
     using Mantid::Kernel::DeltaEMode;
     TestWorkspaceDescriptor wsProps = {
-        5, 10, Environment::SampleOnly, DeltaEMode::Elastic, -1, -1};
-    auto outputWS = runAlgorithm(wsProps);
+        1, 2, false, Environment::CubeRotatedSampleOnly, DeltaEMode::Elastic,
+        -1};
+    auto testWS = setUpWS(wsProps);
+
+    // calculate expected value of the att factor as integral exp(-mu*t)p(t)dt
+    // where t = 2(1-y), p(t) = 2(1-y)
+    // integrate over y= 0 to 1, which gives
+    // E(att) = (1 - exp(-2*mu)*(2*mu+1))/2*mu^2
+    // E(att^2) = (1 - exp(-4*mu)*(4*mu+1))/8*mu^2
+
+    auto mcAbsorb = createAlgorithm();
+    const int NEVENTS = 500000;
+    mcAbsorb->setProperty("EventsPerPoint", NEVENTS);
+
+    TS_ASSERT_THROWS_NOTHING(mcAbsorb->setProperty("InputWorkspace", testWS));
+    TS_ASSERT_THROWS_NOTHING(mcAbsorb->execute());
+    auto outputWS = getOutputWorkspace(mcAbsorb);
 
     verifyDimensions(wsProps, outputWS);
-    const double delta(1e-04);
-    const size_t middle_index(4);
-
-    TS_ASSERT_DELTA(0.6243, outputWS->y(0).front(), delta);
-    TS_ASSERT_DELTA(0.2829, outputWS->y(0)[middle_index], delta);
-    TS_ASSERT_DELTA(0.1110, outputWS->y(0).back(), delta);
-    TS_ASSERT_DELTA(0.6280, outputWS->y(2).front(), delta);
-    TS_ASSERT_DELTA(0.2892, outputWS->y(2)[middle_index], delta);
-    TS_ASSERT_DELTA(0.1178, outputWS->y(2).back(), delta);
-    TS_ASSERT_DELTA(0.6265, outputWS->y(4).front(), delta);
-    TS_ASSERT_DELTA(0.2864, outputWS->y(4)[middle_index], delta);
-    TS_ASSERT_DELTA(0.1143, outputWS->y(4).back(), delta);
+    auto yData = outputWS->getSpectrum(0).dataY();
+    auto eData = outputWS->getSpectrum(0).dataE();
+    // recreate the sd of the set of simulated paths from the sd on the mean
+    double attenuationFactorsSD1 = eData[0] * sqrt(NEVENTS);
+
+    const double delta(1e-03);
+    const double calculatedAttFactor1 = (1 - 3 * exp(-2)) / 2;
+    const double calculatedAttFactorSq1 = (1 - 5 * exp(-4)) / 8;
```

Comment:
The "delta" variable can be. Don't think the other two can though because of the use of the exp() function. There are a few instances of the delta variable so will change all of the them